src/Core/Update.php

```diff
@@ -141,7 +161,15 @@ public static function run(string $basePath, bool $force = false, bool $override
 					Logger::notice('Update starting.', ['from' => $stored, 'to' => $current]);
 
 					// Checks if the build changed during Lock acquiring (so no double update occurs)
-					$retryBuild = DI::config()->get('system', 'build', null);
+					$retryBuild = DI::config()->get('system', 'build');
+					// legacy option - check if there's something in the Config table
+					if (DBStructure::existsTable('config')) {
+						$dbConfig = DBA::selectFirst('config', ['v'], ['cat' => 'system', 'k' => 'build']);
+						if (!empty($dbConfig)) {
+							$retryBuild = $dbConfig['v'];
+						}
+					}
+
```

Comment:
At this point we have a value, ensured in line 144.
```suggestion
```

Reply:
This doesn't work because the next `DI::config()->get('system', 'build') at line 164 retrieves the wrong build number again, so I had to check it here again